### PR TITLE
[Enhancement](java-udf) java-udf module split to sub modules

### LINKED ISSUE #19977: [Enhancement] java-udf module split to sub modules
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

The java-udf module has become increasingly large and difficult to manage, making it inconvenient to package and use as needed. It needs to be split into multiple sub-modules, such as : commom、udf、jdbc-scanner、hudi-scanner、 paimon-scanner.

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by lexluo09**
## Proposed changes

Issue Number: close #19977

The java-udf module has become increasingly large and difficult to manage, making it inconvenient to package and use as needed. It needs to be split into multiple sub-modules, such as : java-commom、java-udf、jdbc-scanner、hudi-scanner、 paimon-scanner.

## Further comments
the java-commom project Includes jni and common function.

### CODE REVIEW

File: docs/zh-CN/docs/faq/install-faq.md [hunk 1]
```diff
@@ -300,8 +300,8 @@ ldd /path/to/libmyodbc8w.so |grep libssl.so
 如果升级后启动 be 出现下面这种 Java `NoClassDefFoundError` 错误
 
 ```
-Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/doris/udf/JniUtil
-Caused by: java.lang.ClassNotFoundException: org.apache.doris.udf.JniUtil
+Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/doris/common/jni/utils/JniUtil
+Caused by: java.lang.ClassNotFoundException: org.apache.doris.common.jni.utils.JniUtil
 ```
 需要从官网下载 `apache-doris-java-udf-jar-with-dependencies-1.2.0` 的 Java UDF 函数依赖包，放到 BE 安装目录下的 lib 目录，然后重新启动 BE
```
Comment (zy-kkk) [inline docs/zh-CN/docs/faq/install-faq.md:306]: Does this jar need to be renamed?

Comment (lexluo09) [reply docs/zh-CN/docs/faq/install-faq.md:306]: en ,yes

**Comment by DongLiang-0** [comment]
You need to copy the entire `be-java-extensions` jar package after successful construction to the `lib` directory of `be` in `build.sh`.

File: be/src/vec/functions/function_java_udf.cpp [hunk 2]
```diff
@@ -38,7 +38,7 @@
 #include "vec/core/block.h"
 #include "vec/data_types/data_type_nullable.h"
 
-const char* EXECUTOR_CLASS = "org/apache/doris/udf/UdfExecutor";
```
Comment (zhangstar333) [inline be/src/vec/functions/function_java_udf.cpp:41]: here seems error, udaf is under be/src/vec/aggregate_functions/aggregate_function_java_udaf.h

Comment (lexluo09) [reply be/src/vec/functions/function_java_udf.cpp:41]: thanks. I will fix it.

**Comment by morningman** [comment]
run buildall

File: fe/be-java-extensions/hudi-scanner/src/main/java/org/apache/doris/hudi/Utils.java [hunk 3]
```diff
@@ -15,9 +15,9 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package org.apache.doris.jni.utils;
+package org.apache.doris.hudi;
 
-import org.apache.commons.io.FileUtils;
+import com.facebook.presto.hadoop.$internal.org.apache.commons.io.FileUtils;
```
Comment (morningman) [inline fe/be-java-extensions/hudi-scanner/src/main/java/org/apache/doris/hudi/Utils.java:20]: Why changing to this?

File: fe/be-java-extensions/paimon-scanner/pom.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        <version>${revision}</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>paimon-scanner</artifactId>
+
+    <properties>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
+        <paimon.version>0.4-SNAPSHOT</paimon.version>
```
Comment (morningman) [inline fe/be-java-extensions/paimon-scanner/pom.xml:17]: change to `0.4.0-incubating`

File: fe/pom.xml [hunk 5]
```diff
@@ -263,16 +263,24 @@ under the License.
...
         <iceberg.version>1.1.0</iceberg.version>
         <maxcompute.version>0.43.3-public</maxcompute.version>
+        <arrow.version>9.0.0</arrow.version>
         <avro.version>1.11.1</avro.version>
         <!-- hudi -->
         <hudi.version>0.13.0</hudi.version>
+        <presto.hadoop.version>2.7.4-11</presto.hadoop.version>
+        <presto.hive.version>3.0.0-8</presto.hive.version>
+        <!-- hudi be scan version TODO Verify compatibility -->
+        <hudi.scan.version>0.12.2</hudi.scan.version>
```
Comment (morningman) [inline fe/pom.xml:282]: should be 0.13?

Comment (lexluo09) [reply fe/pom.xml:282]: yes.

File: fe/be-java-extensions/paimon-scanner/pom.xml (hunk 4 above)
Comment (lexluo09) [reply fe/be-java-extensions/paimon-scanner/pom.xml:17]: yes.

**Comment by morningman** [comment]
run buildall

**Comment by lexluo09** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**